src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -54,20 +57,23 @@
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "The id of the network imported"),
             @ApiResponse(code = 409, message = "The study already exist or the case doesn't exists")})
-    public ResponseEntity<Void> createStudyFromExistingCase(@PathVariable("studyName") String studyName,
-                                                                  @PathVariable("caseUuid") UUID caseUuid,
-                                                                  @RequestParam("description") String description) {
-
-        if (studyService.studyExists(studyName)) {
-            throw new ResponseStatusException(HttpStatus.CONFLICT, STUDY_ALREADY_EXISTS);
-        }
-
-        if (!studyService.caseExists(caseUuid)) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, CASE_DOESNT_EXISTS);
-        }
-
-        studyService.createStudy(studyName, caseUuid, description);
-        return ResponseEntity.ok().build();
+    public Mono<ResponseEntity<Object>> createStudyFromExistingCase(@PathVariable("studyName") String studyName,
+                                                    @PathVariable("caseUuid") UUID caseUuid,
+                                                    @RequestParam("description") String description) {
+        Mono<Boolean> studyExists = studyService.studyExists(studyName);
+        Mono<Boolean> caseExists = studyService.caseExists(caseUuid);
+
+        return Mono.zip(studyExists, caseExists)
+                .flatMap(t -> {
+                    if (t.getT1().equals(Boolean.TRUE)) {
+                        return Mono.just(ResponseEntity.status(HttpStatus.CONFLICT).body(STUDY_ALREADY_EXISTS));
+                    } else if (t.getT2().equals(Boolean.FALSE)) {
+                        return Mono.just(ResponseEntity.status(HttpStatus.NOT_FOUND).body(CASE_DOESNT_EXISTS));
+                    } else {
+                        return studyService.createStudy(studyName, caseUuid, description)
+                                .flatMap(s -> Mono.just(ResponseEntity.ok().build()));
+                    }
+                });
```

Comment:
Maybe it's easier to read if you keep the exceptions instead of returning ResponseEntities and you attach the exceptions to their respective monos (untested)

sequential
```java
studyService.studyExists(studyName).flatmap( exists -> exists ? Mono.error(...) : Mono.empty() )
.then(
  studyService.caseExists(caseUuid).flatmap( exists -> exists ? Mono.empty() : Mono.error(...) )
).then(
studyService.createStudy().thenReturn(ResponseEntity.ok().build())
)
```

I think you get parallel execution by replacing the first ```then``` with ```zipWith```:
```java
studyService.studyExists(studyName).flatmap(...)
.zipWith(
 studyService.caseExists(caseUuid).flatmap(...)
).then(
 studyService.createStudy().thenReturn(...)
)
```
```  .flatmap( exists -> exists ? Mono.error(...) : Mono.empty() )``` can be replaced by ```filter(...).switchOnEmpty(Mono.error())```, not sure which is is cleaner...

Reply:
Mono.error() --> response status 500 INTERNAL EXCEPTION
we want to send the suitable response status 409 conflict or NotFound 404 to know what is the real problem.
ok with the proposition zipWith it's better for a sequential check.

Reply:
or something like
```
Mono.when(assertStudyNotExists, assertCaseExists)
    .then( studyService.createStudy() ) ;
```
?